File: python/src/ev1/ut2_controlflujometodos/u03funciones/teoria/U01_FuncionesIntro.py
"""
U01 — Introduccion a funciones (acciones sin return).

OBJETIVO:
  - Entender que es una funcion y por que sirve modularizar.
  - Definir funciones con def que solo ejecutan acciones (print).
  - Llamar una funcion varias veces desde el programa principal.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
"""


def saludar():
    # Solo hace una accion: imprimir. No usa return.
    print("¡Hola desde la funcion saludar()!")


def imprimir_separador():
    # Reutilizamos este bloque cada vez que queramos una linea visual.
    print("-" * 30)


def contar_hasta_cinco():
    # Dentro de la funcion tambien podemos usar bucles.
    print("Contando hasta 5:", end=" ")
    for i in range(1, 6):
        print(i, end=" ")
    print()


print("\n¡INICIO DEL PROGRAMA!\n")
"""
PRIMERA PARTE - Llamar una funcion varias veces.
"""
print(f"\n¡DEMO 1: SALUDAR!\n")

# Cada llamada ejecuta el cuerpo de saludar() otra vez.
saludar()
saludar()

"""
SEGUNDA PARTE - Funcion que imprime un separador.
"""
print(f"\n¡DEMO 2: SEPARADOR!\n")

imprimir_separador()
print("Contenido entre separadores")
imprimir_separador()

"""
TERCERA PARTE - Funcion con logica interna (bucle).
"""
print(f"\n¡DEMO 3: CONTAR HASTA 5!\n")

contar_hasta_cinco()

print("\n¡FIN DEL PROGRAMA!\n")

File: cpp/src/ev1/ut2_controlflujometodos/u04metodos/teoria/U01_FuncionesIntro.cpp
/*
U01 — Introduccion a funciones void.

OBJETIVO:
  - Definir funciones void que no devuelven valor.
  - Llamar una funcion varias veces desde main.
  - Agrupar codigo repetido en un bloque con nombre.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
using namespace std;

void Saludar()
{
    cout << "Hola desde la funcion Saludar()" << endl;
}

void ImprimirSeparador()
{
    for (int i = 0; i < 30; i++)
    {
        cout << '-';
    }
    cout << endl;
}

void ContarHasta()
{
    cout << "Contando hasta 5: ";
    for (int i = 1; i <= 5; i++)
    {
        cout << i << " ";
    }
    cout << endl;
}

int main() {
    cout << "\n¡INICIO DEL PROGRAMA!\n" << endl;

    /*
    PRIMERA PARTE - Llamar una funcion void varias veces.
      void = no devuelve nada; solo ejecuta acciones (side effects).
    */
    cout << "¡DEMO 1: SALUDAR!\n" << endl;
    Saludar();
    Saludar();

    /*
    SEGUNDA PARTE - Funcion que imprime una linea separadora.
    */
    cout << "\n¡DEMO 2: SEPARADOR!\n" << endl;
    ImprimirSeparador();
    cout << "Contenido entre separadores" << endl;
    ImprimirSeparador();

    /*
    TERCERA PARTE - Funcion con logica interna (bucle).
    */
    cout << "\n¡DEMO 3: CONTAR HASTA 5!\n" << endl;
    ContarHasta();

    cout << "\n¡FIN DEL PROGRAMA!\n" << endl;
    return 0;
}
